ServiceWebsite/ServiceWebsite/ClientApp/src/app/modules/individual-journey/pages/court-building-video/court-building-video.component.spec.ts

```diff
@@ -1,8 +1,33 @@
-import { CanCreateComponent } from '../individual-base-component/component-test-bed.spec';
 import { CourtBuildingVideoComponent } from './court-building-video.component';
+import { Component, Input } from '@angular/core';
+import { VideoUrlService } from '../../services/video-url.service';
+import { IndividualJourney } from '../../individual-journey';
+import { CanCreateVideoViewBaseComponent } from '../../components/video-view-base/video-view-base.component.spec';
 
 describe('CourtBuildingVideoComponent', () => {
-  it('can be created', () => {
-    CanCreateComponent(CourtBuildingVideoComponent);
+  it('can be created', (() => {
+    CanCreateVideoViewBaseComponent(CourtBuildingVideoComponent);
+  }));
+
+  describe('functionality', () => {
+    let component: CourtBuildingVideoComponent;
+    const videoUrlService = jasmine.createSpyObj<VideoUrlService>(['getVideoFileUrl']);
+
+    beforeEach(() => {
+      const journey = new IndividualJourney();
+      component = new CourtBuildingVideoComponent(journey, videoUrlService);
+    });
+
+    it('should set the video source when initialized', () => {
```

Comment:
so you can test these on the base class maybe? then you don't have any code on this actual component to test

Reply:
I have moved in base class.
In my case I can test in the base class, but if I override my base method then test will not catch an error.